File: output/ChatGPT/codenet/Java/Python/s973587235.py
n = int(input())

minN = int(input()) - 1

for i in range(n - 1):
    h = int(input())
    if minN == h:
        minN = h
    elif minN == h - 1:
        minN = h - 1
    elif minN < h - 1:
        minN = h - 1
    elif minN > h:
        print("No")
        exit()

print("Yes")

File: output/ChatGPT/codenet/Java/C++/s973587235.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;
    int minN = 0;
    cin >> minN;
    for (int i = 0; i < n - 1; i++) {
        int h;
        cin >> h;
        if (minN == h) {
            minN = h;
        } else if (minN == h - 1) {
            minN = h - 1;
        } else if (minN < h - 1) {
            minN = h - 1;
        } else if (minN > h) {
            cout << "No" << endl;
            return 0;
        }
    }
    cout << "Yes" << endl;

    return 0;
}